compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -185,30 +148,16 @@ impl<'a> Parser<'a> {
                 }
             }
             NonterminalKind::Path => {
-                let (mut path, tokens) =
-                    self.collect_tokens(|this| this.parse_path(PathStyle::Type))?;
-                // We have have eaten an NtPath, which could already have tokens
-                if path.tokens.is_none() {
-                    path.tokens = tokens;
-                }
+                let path = self.collect_tokens(|this| this.parse_path(PathStyle::Type))?;
                 token::NtPath(path)
             }
             NonterminalKind::Meta => {
-                let (mut attr, tokens) = self.collect_tokens(|this| this.parse_attr_item(false))?;
-                // We may have eaten a nonterminal, which could already have tokens
-                if attr.tokens.is_none() {
-                    attr.tokens = tokens;
-                }
+                let attr = self.collect_tokens(|this| this.parse_attr_item(false))?;
                 token::NtMeta(P(attr))
             }
             NonterminalKind::TT => token::NtTT(self.parse_token_tree()),
             NonterminalKind::Vis => {
-                let (mut vis, tokens) =
-                    self.collect_tokens(|this| this.parse_visibility(FollowedByType::Yes))?;
-                // We may have etan an `NtVis`, which could already have tokens
-                if vis.tokens.is_none() {
-                    vis.tokens = tokens;
-                }
+                let vis = self.collect_tokens(|this| this.parse_visibility(FollowedByType::Yes))?;
                 token::NtVis(vis)
```

Comment:
```suggestion
                token::NtVis(self.collect_tokens(|this| this.parse_visibility(FollowedByType::Yes))?)
```
Style: `fn parse_nonterminal` now contains multiple pointless single-use variables, all of which could be inlined.